File: ST/37_regex_and_pattern_in_programming/examples/Python/ValidateNumbers.py
import re
my_string = "12ab"
if re.fullmatch(r"\d+", my_string):
    print("The string contains only digits!")
else:
    print("The string does not contain only digits.")

# The Number Regex Pattern
# Integers: ^\\d+$
# Decimals: ^\\d+\\.\\d+$
# Formatted Numbers (like 1,000): ^(\\d{1,3}(,\\d{3})*|\\d+)(\\.\\d+)?$

File: ST/37_regex_and_pattern_in_programming/examples/JavaScript/ValidateNumbers.js
function validateNumbers(str) {

    // regex pattern for  numbers
    const re =/^\d+$/;

    // check if the string contains only numbers
    let result = str.match(re);
    if (result) {
        console.log('The string contains only numbers.');
    }
    else {
        let str = prompt('Contains non-digits:');
        validateNumbers(str);
    }
}

// take input
let str = '3 bananas';

validateNumbers(str);